perceval/components/abstract_processor.py

```diff
@@ -686,19 +686,24 @@ def with_input(self, input_state: BasicState) -> None:
         if self._min_detected_photons_filter is None:
             self._deduce_min_detected_photons(expected_photons)
 
-    def flatten(self) -> list[tuple]:
+    def flatten(self, max_iterations = None) -> list[tuple]:
```

Comment:
Your `max_iterations` parameter is controlling the depth until which you'd like to flatten the circuit, isn't it?
If so, could you rename it to `max_depth`?
Also, could you document it in the method's docstring?